storage/dataflux/fast_list_test.go

```diff
@@ -192,3 +200,249 @@ func TestNewLister(t *testing.T) {
 		})
 	}
 }
+
+func TestNextBatchContextCancelEmulated(t *testing.T) {
+	transportClientTest(context.Background(), t, func(t *testing.T, ctx context.Context, project, bucket string, client *storage.Client) {
+
+		bucketHandle := client.Bucket(bucket)
+		if err := bucketHandle.Create(ctx, project, &storage.BucketAttrs{
+			Name: bucket,
+		}); err != nil {
+			t.Fatal(err)
+		}
+		if err := createObject(ctx, bucketHandle, 2, ""); err != nil {
+			t.Fatalf("unable to create objects: %v", err)
+		}
+		c := NewLister(client, &ListerInput{BucketName: bucket})
+		defer c.Close()
+		childCtx, cancel := context.WithCancel(ctx)
+		cancel()
+		result, err := c.NextBatch(childCtx)
+		if err != nil && !strings.Contains(err.Error(), context.Canceled.Error()) {
```

Comment:
@BrennaEpp  for comment "This error check should use errors.Is"

GRPC has context.Canceled.Error() in desc and not as error. So this succeeds for http but fails for GRPC.

Reply:
You can use the status package to check the error without string-matching, for example like this test here: https://github.com/googleapis/google-cloud-go/blob/main/storage/client_test.go#L1428

Reply:
Thanks! Updated it.